src/internal/git/push.go

```diff
@@ -96,6 +96,23 @@ func push(localPath, tunnelUrl string, spinner *message.Spinner) error {
 		return fmt.Errorf("unable to remove unused git refs from the repo: %w", err)
 	}
 
+	// Fetch remote offline refs in case of old update or if multiple refs are specified in one package
+	fetchOptions := &git.FetchOptions{
+		RemoteName: offlineRemoteName,
+		Auth:       &gitCred,
+		RefSpecs: []goConfig.RefSpec{
+			"refs/heads/*:refs/heads/*",
+			onlineRemoteRefPrefix + "*:refs/heads/*",
+			"refs/tags/*:refs/tags/*",
+		},
+	}
+
+	err = repo.Fetch(fetchOptions)
+	if err != nil {
+		spinner.Debugf("%s", err.Error())
```

Comment:
Shouldn't we return the err here instead of trying to push a repo we failed to fetch?

Reply:
If the repo doesn't exist we would expect it to fail.  We could add an explicit catch for that error though, but given we had no catch here earlier I felt it would be best just to log the error for now.

Reply:
I would add an explicit catch earlier and surface the error.  That way it's clear where the failure is an why